src/table/TableWrapper.jsx

```diff
@@ -42,31 +51,26 @@ export default function TableWrapper(props) {
   const setShouldRefocus = () => {
     shouldRefocus.current = rootElement.getElementsByTagName('table')[0].contains(document.activeElement);
   };
-
+  const getBodyHeight = (updatedRootElement) =>
+    updatedRootElement.clientHeight - updatedRootElement.getElementsByTagName('thead')[0]?.clientHeight - 52;
   const handleChangePage = (event, newPage) => {
     setPageInfo({ top: newPage * rowsPerPage, height: rowsPerPage });
     setPage(newPage);
   };
-
-  // should trigger reload
   const handleChangeRowsPerPage = (event) => {
     setRowsPerPage(+event.target.value);
     setPageInfo({ top: 0, height: +event.target.value });
     setPage(0);
   };
 
-  if (!rows.length && page > 0) {
-    handleChangePage(null, 0);
```

Comment:
when selections are made on another page than the first page and rows per page are bigger than the selected rows, in this case, it causes many renders. To reduce the number of re-renders, moving the logic to its parent, component()

Reply:
I see your point. This could be a good improvement, but it is not risk free. So maybe not worth it so close to release

Reply:
i agree. not rush to merge.

Reply:
I like this idea. But there are two problems though. Moving logic to index making hard to test and passing 4 more properties (minor problem). There is another way and that is always setting to first page when confirming selections. But that wouldn't help for outside updates. I'm still for it but maybe someone else should look at it as well.